### ISSUE TITLE
testinfra with ansible : capacity to exclude groups ?

### ISSUE THREAD

**Issue by mneveu-mx** [philpep/testinfra#541]
Hello,

I use pytest/testinfra with ansible module for unitary tests on my linux nodes.
The command I use is like : 
`/usr/local/bin/pytest [... options ...] **--hosts=ansible://group1,ansible://group2**`
For example, group2 can be a subelement of group1.

My problem : I want to execute some tests on group 1 hosts but without the hosts of group 2..
Is there a way to exclude group2 with a specific syntax ?

Note : with Ansible, there is a syntax to exclude a group from a playbook, for example : 
`hosts: group1 !group2`
So maybe it can be the same with testinfra.

Thank you by advance,
Maxime

**Comment by philpep** [comment]
Yes, I think we could implement this using the same ansible syntax as you suggest.

**Comment by xeor** [comment]
I'm looking for almost the same thing. Something I can mark a test with to only include that test for certain groups. Something like `@pytest.mark.ansible_only_group('mygroup')`.

Using parametrize with testinfra is nice, but I end up with hundreds of yellow skip statuses, since I currently need to do a check inside the test like this..

```
    if 'mygroup' not in host.ansible.get_variables()['group_names']:
        pytest.skip()
```
